UM/Settings/ContainerRegistry.py

```diff
@@ -105,153 +155,159 @@ def findContainerStacks(self, **kwargs) -> List[ContainerStack]:
+                    if metadata["id"] in provider.getAllIds(): #This is the one we need to load it from!
+                        new_container = provider.loadContainer(metadata["id"])
+                        self._containers[new_container.getId()] = new_container
+                        self.metadata[new_container.getId()] = new_container.getMetaData() #Should be the same if deserializeMetadata results in the same metadata as deserialize, but link this by reference.
+                        result.append(new_container)
+                        break
+                else:
+                    Logger.log("w", "The metadata of container {container_id} was loaded, but no container provider seems to be able to load the actual container.".format(container_id = metadata["id"]))
+        return result
+
+    ##  Find the metadata of all container objects matching certain criteria.
+    #
+    #   \param container_type If provided, return only objects that are
+    #   instances or subclasses of ``container_type``.
+    #   \param kwargs A dictionary of keyword arguments containing keys and
+    #   values that need to match the metadata. An asterisk can be used to
+    #   denote a wildcard.
+    #   \return A list of metadata dictionaries matching the search criteria, or
+    #   an empty list if nothing was found.
+    def findContainersMetadata(self, *, ignore_case = False, **kwargs) -> List[Dict[str, Any]]:
+        #Create the query object.
+        query = ContainerQuery.ContainerQuery(self, ignore_case = ignore_case, **kwargs)
+
+        if "id" in kwargs:
+            if kwargs["id"] not in self.metadata: #If we're looking for an unknown ID, try to lazy-load that one.
+                for provider in self._providers:
+                    if kwargs["id"] in provider.getAllIds():
+                        metadata = provider.loadMetadata(kwargs["id"])
+                        self.metadata[metadata["id"]] = metadata
+                        break #Just need to find the first one.
+
+            if query.isIdOnly(): #If we are just searching for a single container by ID, look it up from the ID-based cache.
+                if kwargs["id"] in self.metadata:
+                    return [self.metadata[kwargs["id"]]]
+                else:
+                    return [] #No result, so return an empty list.
 
         if query in self._query_cache:
-            # If the exact same query is in the cache, we can re-use the query result
-            self._query_cache.move_to_end(query) # Query was used, so make sure to update its position
+            #If the exact same query is in the cache, we can re-use the query result.
+            self._query_cache.move_to_end(query) #Query was used, so make sure to update its position so that it doesn't get pushed off as a rarely-used query.
             return self._query_cache[query].getResult()
 
-        # Execute the query, then add it to the cache
         query.execute()
```

Comment:
If "id" is in kwargs, we could provide the candidate list (consisting of max 1 item). Then providing id and some other property will be almost equally fast as id only.